pkg/cmd/infra/router/template.go

```diff
@@ -287,6 +293,83 @@ func parseCaptureHeaders(in string) ([]templateplugin.CaptureHTTPHeader, error)
 	return captureHeaders, nil
 }
 
+func parseHeadersToBeSetOrDeleted(in string) ([]templateplugin.HTTPHeader, error) {
+	var captureHeaders []templateplugin.HTTPHeader
+	var capture templateplugin.HTTPHeader
+	var err error
+	if len(in) == 0 {
+		return captureHeaders, fmt.Errorf("encoded header string not present.")
+	}
+	if len(in) > 0 {
+		for _, header := range strings.Split(in, ",") {
+			parts := strings.Split(header, ":")
+			num := len(parts)
+			switch num {
+			default:
+				return captureHeaders, fmt.Errorf("invalid HTTP header input specification: %v", header)
+			case 3:
+				{
+					err = checkValidHeaderName(parts[0])
+					if err != nil {
+						return captureHeaders, err
+					}
+					headerValue, err := url.QueryUnescape(parts[1])
+					if err != nil {
+						return captureHeaders, fmt.Errorf("failed to decode percent encoding: %v", parts[1])
+					}
+					sanitizedHeaderValue := templateplugin.SanitizeHeaderValue(headerValue)
+					err = checkValidAction(parts[2])
+					if err != nil {
+						return captureHeaders, err
+					}
+					capture = templateplugin.HTTPHeader{
+						Name:   parts[0],
+						Value:  sanitizedHeaderValue,
+						Action: routev1.RouteHTTPHeaderActionType(parts[2]),
+					}
+					captureHeaders = append(captureHeaders, capture)
+				}
+			case 2:
+				{
+					err = checkValidHeaderName(parts[0])
+					if err != nil {
+						return captureHeaders, err
+					}
+					err = checkValidAction(parts[1])
+					if err != nil {
+						return captureHeaders, err
+					}
+					capture = templateplugin.HTTPHeader{
+						Name:   parts[0],
+						Action: routev1.RouteHTTPHeaderActionType(parts[1]),
+					}
+					captureHeaders = append(captureHeaders, capture)
+				}
+			}
+		}
+	}
+
+	return captureHeaders, err
+}
+
+func checkValidAction(action string) error {
+	if action != "Set" && action != "Delete" {
```

Comment:
Don't use strings here, use the types, something like this:
```suggestion
	if action != string(routev1.Set) && action != string(routev1.Delete) {
```

Reply:
done